### DISCUSSION TITLE
fheroes2 port

### DISCUSSION THREAD

**Discussion by rockweather** [OnionUI/Onion#1708] [Ideas]
Is there any chance to port fheroes2? Do you consider to do this in the future?
Also, I have a little bit of experience in coding (mostly simple scripts), how hard is it to do if I decide to dabble in it myself? What do I need to know to do this, and where do I start to learn?

**Comment by rockweather** [comment]
I've successfully ported fheroes2 to Miyoo Mini Plus, the device is running the main menu, the intro videos and the music, but now I'm stuck with implementing mouse control by DPad (I do not know C++ on such level as to clearly understand internal fheroes2 logic). If anyone is interested to help, it will be really appreciated.

**Reply by JEnric** [reply]
Did you implement mouse control? I just tried heroes2 on dosbox pure and the performance is only 90%. Audio is lagging and bit especially in Multiplayer between two miyoo mini plus.

**Reply by rockweather** [reply]
Yes, but only partially. I've remapped miyoo buttons to keyboard keys, created a cursor class method which took key inputs and sent them to cursor movement method (or something like that). There was a problem with screen refresh logic, the cursor was doubling during movement in zero point (upper left corner), it's movement was jittery and cursor itself was flickering. I remember trying different approaches, but couldn't find the solution at that point. I'm not a programmer, sadly.
